cmd/listen.go

```diff
@@ -118,16 +110,28 @@ func (l *Listen) handlePackets(s *SendPktFeed, wg *sync.WaitGroup) {
 				log.Errorf("%s: Unable to decode: %s", l.iname, errx.Error())
 			}
 
+			// if our interface is non-promisc, learn the client IP
+			if l.promisc {
+				l.learnClientIP(packet)
+			}
+
 			log.Debugf("%s: received packet and fowarding onto other interfaces", l.iname)
 			s.Send(packet, l.iname, l.handle.LinkType())
 		case <-ticker: // our timer
 			log.Debugf("handlePackets(%s) ticker", l.iname)
+			// clean client cache
```

Comment:
using the ticker for this is kinda ugly... but it works just fine for the needs.  probably not worth fixing anytime soon?